Patch-Source: https://github.com/organicmaps/organicmaps/pull/7982
---
From 905b161487865a200ce4a43fb2c1d96ecb38e102 Mon Sep 17 00:00:00 2001
From: Gerion Entrup <gerion.entrup@flump.de>
Date: Thu, 25 Apr 2024 10:14:41 +0200
Subject: [PATCH] CMakeLists: import jansson with pkgconfig

Upstream jansson offers compiling with autotools and CMake.
If it is compiled with autotools (the recommended way) it does only
provide a pkgconfig file and not a cmake file.
This is also the way most distributions provide jansson, so this commit
changes the behavior to use pkgconfig, too.

Signed-off-by: Gerion Entrup <gerion.entrup@flump.de>
---
 CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1902f0e618119..975a402f8adc2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -320,8 +320,11 @@ if (WITH_SYSTEM_PROVIDED_3PARTY)
   set(GFLAGS_USE_TARGET_NAMESPACE ON)
   find_package(gflags REQUIRED)
 
+  find_package(PkgConfig)
+  pkg_check_modules(jansson REQUIRED IMPORTED_TARGET jansson)
+  add_library(jansson::jansson ALIAS PkgConfig::jansson)
+
   find_package(expat CONFIG REQUIRED)
-  find_package(jansson CONFIG REQUIRED)
   find_package(pugixml REQUIRED)
   find_package(utf8cpp REQUIRED)
 endif()
